### PR TITLE
Improve platform inference in shell initialisation.

### PR DESCRIPTION
**Pull request by marc0der**
This is the first of multiple PRs to enable support for the new ARM-based candidate versions. This PR is dependent on some serverside work being done first.

### CODE REVIEW

File: src/main/bash/sdkman-init.sh [hunk 1]
```diff
@@ -30,24 +30,67 @@ if [ -z "$SDKMAN_DIR" ]; then
...
+	kernel="$(uname -s)"
+	machine="$(uname -m)"
+	
+	case $kernel in
+	Linux)
+	  case $machine in
+	  i686)
+		echo "Linux32"
+		;;
+	  X86_64)
```
Comment (pago88) [inline src/main/bash/sdkman-init.sh:47]: I'm not sure if this is the case always, but in my system i see the x lowercase: x86-64. Not the end of the world cause the default case catches it, but I still think it should be lowercase to actually match the correct string

**Review by pago88 — CHANGES_REQUESTED** [review]

File: src/main/bash/sdkman-init.sh (hunk 1 above)
Comment (marc0der) [reply src/main/bash/sdkman-init.sh:47]: Thanks for the feedback @pago88. Would you be able to paste your exact shell output for `uname -s` and `uname -m` please? Also, could I ask you to please drop me a dump of `neofetch` if you have it installed?

Comment (pago88) [reply src/main/bash/sdkman-init.sh:47]: I'm running/coding sdkman inside a docker container, so uname -s is Linux and uname -m is x86_64 (lowercase x). For what it's worth, my host is an intel mac, and uname -m also returns x86_64, lowercase x
Result of neofetch on the container:

       _,met$$$$$gg.          root@759d2e7b5100 
    ,g$$$$$$$$$$$$$$$P.       ----------------- 
  ,g$$P"     """Y$$.".        OS: Debian GNU/Linux 10 (buster) x86_64 
 ,$$P'              `$$$.     Host: BHYVE 1.0 
',$$P       ,ggs.     `$$b:   Kernel: 4.19.121-linuxkit 
`d$$'     ,$P"'   .    $$$    Uptime: 5 hours, 46 mins 
 $$P      d$'     ,    $$P    Packages: 233 (dpkg) 
 $$:      $$.   -    ,d$$'    Shell: bash 5.0.3 
 $$;      Y$b._   _,d$P'      CPU: Intel i5-5257U (2) @ 2.698GHz 
 Y$$.    `.`"Y$$$$P"'         Memory: 1274MiB / 1989MiB 
 `$$b      "-.__
  `Y$$                                                
   `Y$$.
     `$$b.
       `Y$$b.
          `"Y$b._
              `"""

Comment (marc0der) [reply src/main/bash/sdkman-init.sh:47]: Great stuff, thanks for taking the time. I'll make that an explicit check.

File: src/test/groovy/sdkman/specs/PlatformSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		where:
+		kernel 		| machine	| platform
+		"Linux"		| "i686"	| "Linux32"
+		"Linux" 	| "X86_64"	| "Linux64"
+		"Linux"		| "armv7l"	| "LinuxARM32"
+		"Linux"		| "armv8l"	| "LinuxARM64"
+		"Linux"		| "aarch64"	| "LinuxARM64"
+		"Linux"     | ""	    | "Linux64"
+		"Darwin"	| "X86_64"	| "Darwin64"
+		"Darwin"	| "arm64"	| "DarwinARM6"
```
Comment (eddumelendez) [inline src/test/groovy/sdkman/specs/PlatformSpec.groovy:30]: missing a `4` at the platform

File: src/main/bash/sdkman-init.sh [hunk 3]
```diff
@@ -30,24 +30,67 @@ if [ -z "$SDKMAN_DIR" ]; then
...
+		echo "LinuxARM64"
+		;;
+	  *)
+	  	echo "Linux64"
+	  	;;
+	  esac
+	  ;;
+	Darwin)
+	  case $machine in
+	  X86_64)
```
Comment (mecca831) [inline src/main/bash/sdkman-init.sh:66]: Similar to the comment above, the output from my Intel MBP is:

```
$ uname -m
x86_64
```

File: src/main/bash/sdkman-init.sh (hunk 1 above)
Comment (marc0der) [reply src/main/bash/sdkman-init.sh:47]: That's been fixed. Thanks!

File: src/test/groovy/sdkman/specs/PlatformSpec.groovy (hunk 2 above)
Comment (marc0der) [reply src/test/groovy/sdkman/specs/PlatformSpec.groovy:30]: Done!

**Comment by marc0der** [comment]
I've merged this with CLI, but Github seems a bit confused. Closing this PR manually.

**Pull request closed without merging**